src/pages/workspace/withPolicyConnections.tsx

```diff
@@ -27,18 +27,20 @@ function withPolicyConnections<TProps extends WithPolicyConnectionsProps>(Wrappe
         const [hasConnectionsDataBeenFetched, {status}] = useOnyx(`${ONYXKEYS.COLLECTION.POLICY_HAS_CONNECTIONS_DATA_BEEN_FETCHED}${props.policy?.id ?? '0'}`, {
             initWithStoredValues: false,
         });
+        // eslint-disable-next-line rulesdir/no-negated-variables
+        const isConnectionDataFetchNotNeeded = isOffline || !props.policy || !props.policy.areConnectionsEnabled || !!hasConnectionsDataBeenFetched || !!props.policy.connections;
 
         useEffect(() => {
             // When the accounting feature is not enabled, or if the connections data already exists,
             // there is no need to fetch the connections data.
-            if (!props.policy || !props.policy.areConnectionsEnabled || !!hasConnectionsDataBeenFetched || !!props.policy.connections) {
+            if (isConnectionDataFetchNotNeeded || !props?.policy?.id) {
                 return;
             }
 
             openPolicyAccountingPage(props.policy.id);
-        }, [hasConnectionsDataBeenFetched, props.policy, isOffline]);
+        }, [hasConnectionsDataBeenFetched, props.policy, isOffline, isConnectionDataFetchNotNeeded]);
 
-        if (props.policy?.areConnectionsEnabled && (!props.policy || status === 'loading' || hasConnectionsDataBeenFetched === false)) {
```

Comment:
After removing the `hasConnectionsDataBeenFetched` condition, the loader won't show on any page?

Reply:
status is connected to hasConnectionsDataBeenFetched in onyx. Initially hasConnectionsDataBeenFetched - will be null and status = 'loading' - probably if any of our pages which use current HOC - will ask for  hasConnectionsDataBeenFetched - FullScreenLoading indicator will be shown

Reply:
Since `initWithStoredValues` is `false`, the status will be `loaded`.